python/tvm/micro/model_library_format.py

```diff
@@ -277,6 +294,37 @@ def _create_empty_entry(target_device_type):
             main_func_metadata.io_sizes[target]
         )
 
+        # Now, we also add the information about the size of each input and output of the main
+        # function (in bytes)
+        input_dict = {}
+        for input_param in main_func_metadata.relay_primfuncs[target].params:
+            if hasattr(input_param, "checked_type"):
+                input_dict[input_param.name_hint] = int(
+                    _shape_to_size(input_param.checked_type.shape, input_param.checked_type.dtype)
+                )
+            else:
+                # TODO: maybe fill checked_type here?
+                input_dict[input_param.name_hint] = 0
+        target_main_entries[int(target.kind.device_type)]["inputs"] = input_dict
+
+        output_dict = {}
+        # For output, we dont have the name of the output, so we enumerate them
+        if isinstance(main_func_metadata.relay_primfuncs[target].ret_type, tvm.ir.type.TupleType):
```

Comment:
Given tuples could be more heavily nested than this, we should use some recursive function to generate the outputs? We can re-purpose this function (which is no longer used): https://github.com/apache/tvm/blob/main/python/tvm/micro/model_library_format.py#L293

Reply:
Applied the proposed change